src/ResourceManager/EventHub/Commands.EventHub/Cmdlets/AuthorizationRule/SetAzureEventhubAuthorizationRules.cs

```diff
@@ -0,0 +1,117 @@
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.EventHub.Models;
+using Microsoft.Azure.Management.EventHub.Models;
+using System.Management.Automation;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.Commands.EventHub.Commands
+{
+    /// <summary>
+    /// 'Set-AzureRmRelayAuthorizationRule' Cmdlet updates the specified AuthorizationRule
+    /// </summary>
+    [Cmdlet(VerbsCommon.Set, EventHubAuthorizationRuleVerb, DefaultParameterSetName = NamespaceAuthoRuleParameterSet, SupportsShouldProcess = true), OutputType(typeof(SharedAccessAuthorizationRuleAttributes))]
+    public class SetAzureEventhubAuthorizationRules : AzureEventHubsCmdletBase
+    {
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 0,
+            HelpMessage = "Resource Group Name.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 1, ParameterSetName = NamespaceAuthoRuleParameterSet,
+            HelpMessage = "Namespace Name.")]
+        [Parameter(Mandatory = true, Position = 1, ParameterSetName = EventhubAuthoRuleParameterSet)]
+        [ValidateNotNullOrEmpty]
+        [Alias(AliasNamespaceName)]
+        public string Namespace { get; set; }
+
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 2, ParameterSetName = EventhubAuthoRuleParameterSet,
+            HelpMessage = "EventHub Name.")]
+        [ValidateNotNullOrEmpty]
+        [Alias(AliasEventHubName)]
+        public string EventHub { get; set; }
+
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 3,
+            HelpMessage = "AuthorizationRule Name.")]
+        [ValidateNotNullOrEmpty]
+        [Alias(AliasAuthorizationRuleName)]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = true,
+            ValueFromPipeline = true,
+            Position = 4,
+            ParameterSetName = AuthoRuleInputObjectParameterSet,
+            HelpMessage = "Relay AuthorizationRule Object.")]
+        [Parameter(ParameterSetName = NamespaceAuthoRuleParameterSet)]
```

Comment:
@v-Ajnava please make sure parameters have the same attributes across the different parameter sets. Right now `InputObject` is optional, has no position, and cannot accept input from the pipeline in the `NamespaceAuthoRuleParameterSet` and `EventHubAuthoRuleParameterSet` since those values are not set